### PR TITLE
Comply with rubocop (rails.yml)

### PR DESCRIPTION
**Pull request by avinhurry**
### Context

We have lots of tech debt because of cops not being enabled. This PR enables some of them and fixes the bad code. 

**This PR focuses on clearing out the `rails.yml` file.**

### Changes proposed in this pull request

See commits

### Guidance to review

- Recommend reviewing commit by commit, as in each commit  one particular cop is enabled and all the code is adjusted to pass the linter.

- Do we want to keep any of these cops disabled?

- Play around with the review app in areas which relies on lots of code working together to ensure there are no (untested) hidden breakages.

- Check everywhere (Find/Publish/Support)

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by avinhurry** [comment]
[Comply with rubocop (rails.yml)](https://trello.com/c/8ZHigXvk/1011-comply-with-rubocop-railsyml)

File: Rakefile [hunk 1]
```diff
@@ -10,7 +10,7 @@ Rails.application.load_tasks
 if %w[development test].include? Rails.env
   task lint: %w[lint:ruby lint:erb lint:js]
   task parallel: ['parallel:spec']
-  task :javascript_specs do
+  task javascript_specs: :environment do
```
Comment (avinhurry) [inline Rakefile:13]: Does this look right?

File: app/models/course.rb [hunk 2]
```diff
@@ -156,7 +156,7 @@ def new_draft_attributes
   end
 
   has_one :latest_published_enrichment, -> { published.order('created_at DESC, id DESC').limit(1) },
-          class_name: 'CourseEnrichment'
+          class_name: 'CourseEnrichment', inverse_of: :course
```
Comment (avinhurry) [inline app/models/course.rb:159]: Does this look right?

**Review by defong — APPROVED** [review]
LGTM

**Pull request merged**